mongo/src/main/java/com/redhat/lightblue/mongo/crud/Translator.java

```diff
@@ -431,6 +433,29 @@ public DBObject translateProjection(EntityMetadata md,
     }
 
     /**
+     * Populate mongo controlled hidden fields that may exist
+     *
+     * @param md
+     * @param dbo
+     * @return
+     */
+    public static DBObject populateHiddenFields(EntityMetadata md, DBObject dbo) {
+        md.getEntityInfo().getIndexes().getCaseInsensitiveIndexes().stream()
+            .map(IndexSortKey::getField)
+            .forEach(p -> {
+                Path hiddenPath = getHiddenForField(p);
+                Object object = dbo.get(p.toString());
+                if(object instanceof String) {
```

Comment:
This won't work if p has multiple segments. You need to descend down the dbo tree. There should be a function somewhere in translator that does that